includes/class-sensei-quiz.php

```diff
@@ -2420,19 +2436,30 @@ private static function maybe_get_button_html_for_quiz_footer( $lesson_id = null
 			return null;
 		}
 
-		$lesson_status = \Sensei_Utils::user_lesson_status( $lesson_id, $user_id );
+		$quiz_id       = (int) Sensei()->lesson->lesson_quizzes( $lesson_id );
+		$quiz_progress = Sensei()->quiz_progress_repository->get( $quiz_id, $user_id );
+		if ( ! $quiz_progress ) {
+			return null;
+		}
 
-		$is_complete = $lesson_status && in_array( $lesson_status->comment_approved, [ 'complete', 'graded', 'passed', 'failed' ], true );
+		$is_quiz_complete = in_array(
```

Comment:
Maybe extract it as a method of quiz progress? (If it makes sense, of course.) Like this one:
https://github.com/Automattic/sensei/blob/trunk/includes/internal/student-progress/quiz-progress/models/class-quiz-progress-abstract.php#L213C25-L213C25

Reply:
I was debating with myself if it makes sense, but now that you mention it... I've added the helper method in 1baf3dbba931e6c1825faff8fc7467f3b5778036.